components/autocomplete/index.js

```diff
@@ -330,11 +338,25 @@ export class Autocomplete extends Component {
 		// create a regular expression to filter the options
 		const search = open ? new RegExp( '(?:\\b|\\s|^)' + escapeRegExp( query ), 'i' ) : /./;
 		// filter the options we already have
-		const filteredOptions = open ? filterOptions( search, this.state[ 'options_' + open.idx ] ) : [];
+		let filteredOptions = open ? filterOptions( search, this.state[ 'options_' + open.idx ] ) : [];
```

Comment:
If you rename the asynchronously calculated output further down this doesn't need to be a `let`.